doc/manual/src/language/index.md

```diff
@@ -434,7 +434,19 @@ This is an incomplete overview of language features, by example.
   </td>
   <td>
 
-   Add all attributes from the given set to the scope (evaluates to `1`)
+   Add all attributes from the given set to the scope (evaluates to `1`).
+
+   **Warning:**
+   In contrast to `inherit`, does not shadow existing variables with the same name!
+   Existing outer variables get precedence over variables coming from `with`. Example:
+   ```
+   let
+     a = "outer";
+     attrs = { a = "inner"; b = 3; };
+   in
+     with attrs; [ a b ]
+   ```
+   Evaluates to `[ "outer" 3 ]`.
```

Comment:
````suggestion
   **Warning:**
   In contrast to `inherit`, `with` does not shadow existing variables with the same name!
   Outer variables not declared in `with` get precedence over variables coming from `with`.
   ```nix
   let
     a = "outer";
     attrs = { a = "inner"; b = 3; };
   in
     with attrs; [ a b ]
   ```
   Evaluates to `[ "outer" 3 ]`.

   Variables coming from outer `with` expressions *are* shadowed:
   ```nix
   with { a = "outer"; };
   with { a = "inner"; };
   a
   ```
   Does evaluate to `"inner"`.
````